launchpod/src/test/java/com/google/launchpod/FormHandlerServletTest.java

```diff
@@ -0,0 +1,403 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.security.interfaces.DSAKey;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.launchpod.servlets.FormHandlerServlet;
+import com.google.launchpod.data.UserFeed;
+import com.google.launchpod.data.RSS;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.repackaged.com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/** 
+ * Runs unit tests for the FormHandlerServlet that contains doPost(), doGet(), and xmlString() methods.
+*/
+@RunWith(JUnit4.class)
+public class FormHandlerServletTest extends Mockito {
+
+    @InjectMocks
+    private FormHandlerServlet servlet = new FormHandlerServlet();
+
+    @Mock
+    HttpServletRequest request;
+
+    @Mock
+    HttpServletResponse response;
+
+    @Rule // JUnit 4 uses Rules for testing specific messages
+    public ExpectedException thrown = ExpectedException.none();
+
+    private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig()
+    .setDefaultHighRepJobPolicyUnappliedJobPercentage(100));
+
+    // keys
+    private static final String USER_FEED = "UserFeed";
+    private static final String PODCAST_TITLE = "title";
+    private static final String XML_STRING = "xmlString";
+    private static final String MP3_LINK = "mp3link"; // URL to existing MP3 file
+    private static final String TIMESTAMP = "timestamp";
+    private static final String EMAIL = "email"; 
+    private static final String ID = "id";
+
+    private static final String TEST_PODCAST_TITLE = "TEST_PODCAST_TITLE";
+    private static final String TEST_EMPTY_PODCAST_TITLE = "";
```

Comment:
for these constants that are only referenced in one test, inlining them is easier to read. same with other constants that only pertain to one test